src/pykeen/models/base.py

```diff
@@ -276,6 +278,105 @@ def predict_scores_all_tails(
             scores = torch.sigmoid(scores)
         return scores
 
+    def predict_heads(
+        self,
+        relation_label: str,
+        tail_label: str,
+        add_novelties: bool = True,
+        remove_known: bool = False,
+    ) -> pd.DataFrame:
+        """Predict tails for the given head and relation (given by label).
+
+        :param relation_label: The string label for the relation
+        :param tail_label: The string label for the tail entity
+        :param add_novelties: Should the dataframe include a column denoting if the ranked head entities correspond
+         to novel triples?
+        :param remove_known: Should non-novel triples (those appearing in the training set) be shown with the results?
+         On one hand, this allows you to better assess the goodness of the predictions - you want to see that the
+         non-novel triples generally have higher scores. On the other hand, if you're doing hypothesis generation, they
+         may pose as a distraction.
+
+        The following example shows that after you train a model on the Nations dataset,
+        you can score all entities w.r.t a given relation and tail entity.
+
+        >>> from pykeen.pipeline import pipeline
+        >>> result = pipeline(
+        ...     dataset='Nations',
+        ...     model='RotatE',
+        ... )
+        >>> df = result.model.predict_heads('accusation', 'brazil')
+        """
+        tail_id = self.triples_factory.entity_to_id[tail_label]
+        relation_id = self.triples_factory.relation_to_id[relation_label]
+        rt_batch = torch.LongTensor([[relation_id, tail_id]])
+        scores = self.predict_scores_all_heads(rt_batch)
+        rv = pd.DataFrame(
+            [
+                (entity_id, entity_label, scores[:, entity_id].item())
+                for entity_label, entity_id in self.triples_factory.entity_to_id.items()
+            ],
+            columns=['head_id', 'head_label', 'score'],
+        ).sort_values('score', ascending=False)
+        if add_novelties or remove_known:
+            rv['novel'] = rv['head_id'].map(lambda head_id: self._novel(head_id, relation_id, tail_id))
+        if remove_known:
+            rv = rv[rv['novel']]
+            del rv['novel']
+        return rv
+
+    def predict_tails(
+        self,
+        head_label: str,
+        relation_label: str,
+        add_novelties: bool = True,
+        remove_known: bool = False,
+    ) -> pd.DataFrame:
```

Comment:
Since these are user-facing functions, I value this redundancy to make it absolutely clear how the function should be used. I don't think it will be commonly used the third way, but when someone finally asks for it we can add `predict_relations` as well.